### PR TITLE
test (azure-iot-device) : DPS cert management client cert issuance tests

### PR DESCRIPTION
**Pull request by olivakar**
Some portions of provisioning service SDK which was used for tests have been utilized for the tests.
Our dependency is now only on ms-rest.

### CODE REVIEW

File: azure-iot-device/samples/dps-cert-mgmt/provision_symmetric_key_client_cert_issuance.py [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+            msg = Message("test wind speed " + str(i))
+            msg.message_id = uuid.uuid4()
+            await device_client.send_message(msg)
+            print("done sending message #" + str(i))
+
+        # send `messages_to_send` messages in parallel
+        await asyncio.gather(*[send_test_message(i) for i in range(1, messages_to_send + 1)])
+
+        # finally, disconnect
+        await device_client.disconnect()
```
Comment (cartertinney) [inline azure-iot-device/samples/dps-cert-mgmt/provision_symmetric_key_client_cert_issuance.py:76]: Need to use .shutdown() instead

File: tests/e2e/provisioning_e2e/tests/test_sync_symmetric_enrollments.py [hunk 2]
```diff
@@ -6,31 +6,28 @@
 
 from provisioning_e2e.service_helper import Helper, connection_string_to_hostname
 from azure.iot.device import ProvisioningDeviceClient
-from azure.iot.device.common import X509
-from provisioningserviceclient import ProvisioningServiceClient, IndividualEnrollment
-from provisioningserviceclient.protocol.models import AttestationMechanism, ReprovisionPolicy
+from ..provisioningservice.protocol.models import (
```
Comment (cartertinney) [inline tests/e2e/provisioning_e2e/tests/test_sync_symmetric_enrollments.py:9]: Consider just importing `models` and then referring to the specific models through that module to cut down on unnecessary imports. Here and everywhere.

File: tests/e2e/provisioning_e2e/tests/test_sync_symmetric_enrollments.py [hunk 3]
```diff
@@ -6,31 +6,28 @@
...
 PROVISIONING_HOST = os.getenv("PROVISIONING_DEVICE_ENDPOINT")
 ID_SCOPE = os.getenv("PROVISIONING_DEVICE_IDSCOPE")
 service_client = ProvisioningServiceClient.create_from_connection_string(
     os.getenv("PROVISIONING_SERVICE_CONNECTION_STRING")
 )
-device_registry_helper = Helper(os.getenv("IOTHUB_CONNECTION_STRING"))
-linked_iot_hub = connection_string_to_hostname(os.getenv("IOTHUB_CONNECTION_STRING"))
-# TODO Delete this line. This is a pre created variable in key vault now.
-symmetric_key_for_cert_management = os.getenv("DPS_CERT_ISSUANCE_SYM_KEY_SYNC")
+connection_string = os.getenv("IOTHUB_CONNECTION_STRING")
```
Comment (cartertinney) [inline tests/e2e/provisioning_e2e/tests/test_sync_symmetric_enrollments.py:27]: Outside the scope of this PR, but at some point we should probably try and overhaul these DPS E2Es to use some of the enhancements and shared infrastructure Bert created for IoTHub E2Es + stress tests.

Probably can wait til V3 though

**Review by cartertinney — COMMENTED** [review]
Is there a reason this newly generated service client is here rather than in the `dev_utils` package? Feels more appropriate to be there, given the size of the thing.

**Comment by anthonyvercolano** [comment]
Ship it.

**Review by anthonyvercolano — APPROVED** [review]

**Pull request merged**